OpenRA.Game/Widgets/WorldInteractionControllerWidget.cs

```diff
@@ -88,18 +110,27 @@ public override bool HandleMouseInput(MouseInput mi)
 					}
 					else if (dragStart.HasValue)
 					{
-						var newSelection = SelectActorsInBox(World, dragStart.Value, xy, _ => true);
+						var newSelection = DeadzoneSelectBox(World, dragStart.Value, xy, _ => true);
 						World.Selection.Combine(World, newSelection, mi.Modifiers.HasModifier(Modifiers.Shift), dragStart == xy);
 					}
+				} else
+				{
+					if (useClassicMouseStyle)
+						ApplyOrders(World, xy, mi);
 				}
 
 				dragStart = dragEnd = null;
 				YieldMouseFocus(mi);
 			}
 
-			// don't issue orders while selecting
-			if (mi.Button == MouseButton.Right && mi.Event == MouseInputEvent.Down && !hasBox)
-				ApplyOrders(World, xy, mi);
+			if (mi.Button == MouseButton.Right && mi.Event == MouseInputEvent.Down)
+			{
+				if (useClassicMouseStyle)
+					World.Selection.Clear();
+
+				if (!hasBox) // don't issue orders while selecting
```

Comment:
or maybe the other way around: you want to make sure that this will always be called if `!useClassicMouseStyle`.